prime-router/metadata/hl7_mapping/datatypes/contactPoint/XTN.yml

```
@@ -0,0 +1,56 @@
+constants:
+  hl7TelecomPath: '%{hl7TelecomField}(%{telecomIndex})'
+
+elements:
+
+  - name: xtn-telephone-number-from-extension
+    value: [ '%resource.extension(%`rsext-xtn1-telephone-number`).value' ]
+    hl7Spec: [ '%{hl7TelecomPath}-1' ]
+
+  - name: xtn-telecom-use-code-from-extension
+    value: [ '%resource.extension(%`rsext-xtn2-telecom-use-code`).value' ]
```

Do we really need to exclusively use the extension instead of `ContactPoint.use`? It seems like in the majority of the use cases, we could use `use` without issue, it is only in cases like PID.13/14 that `use` could be misleading from the HL7 input.

I'm thinking something like this:
```suggestion
    value: [ '%resource.extension(%`rsext-xtn2-telecom-use-code`).value', '%resource.use' ]
```
or having separate elements with conditions, basically just to say to only use `%resource.use` if `xtn2-telecom-use-code` doesn't exist.

---

The value from HL7 maps to a different value for `use` based on this [sheet](https://docs.google.com/spreadsheets/d/1EE3a5UIGXLi8wMDr49fWTEVx8QS2GFqYgitE3vLOQTQ/edit#gid=0). The value doesn't go directly into `use` so we can't use it for the FHIR to HL7

---

The `valueSet` strategy would also open us up to changing data. For example, if a sender sent PRS in PID.13.2 then `use` would be sent to home since that's what the PID[Patient] mapping indicates. Then on the FHIR to HL7 we would look at the `use` value of home and map PID.13.2 to PRN.

---

Right, we would still want to give priority to the extension for that reason, but in the absence of the extension, I think we'd still want to use `use` with a `valueSet`.

---

When the extension is missing that means the field was blank in the incoming HL7. Would we want to add a value when there wasn't one before? That still feels like changing data

---

I'm good with resolving this based on our conversation this morning! Just for the record.